test/unit/test.xpi.js

```diff
@@ -509,4 +510,21 @@ describe("lib/xpi", function() {
     })
     .catch(done);
   });
+
+  it("Warn when executed against a WebExtension", function(done) {
+    process.chdir(webextensionUnsupportedPath);
+
+    // In this test scenario, jpm is executed agains a WebExtension that contains
+    // both a package.json and a manifest.json files in the root dir, probably because
+    // the add-on is being built from an npm package.
+    var manifest = require(path.join(webextensionUnsupportedPath, "package.json"));
+    xpi(manifest).then(function () {
```

Comment:
it would be better to return this promise object (and remove `done` from the function arguments) so that throwing an exception here will fail the promise chain and cause a real test failure.